### ISSUE TITLE
Error while running Docker Compose build

### ISSUE THREAD

**Issue by TuxADN** [chaskiq/chaskiq#797]
### Discussed in https://github.com/chaskiq/chaskiq/discussions/795

<div type='discussions-op-text'>

<sup>Originally posted by **TuxADN** August 18, 2022</sup>
Hello friends,
Here I'm trying to install the application in a Docker container, so I'm trying to follow the (not detailed and difficult to follow) guide:

https://dev.chaskiq.io/getting-started/docker-for-development

Starts fine but then some errors appear and the installation process exits:

root@:~/chaskiq# docker-compose build

 Building 113.4s (9/18)
 => [internal] load build definition from Dockerfile.development                                                                  0.1s
 => => transferring dockerfile: 44B                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                 0.1s
 => => transferring context: 35B                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/ruby:2.7.5-slim-buster                                                         0.3s
 => [internal] load build context                                                                                                 0.0s
 => => transferring context: 216B                                                                                                 0.0s
 => [ 1/14] FROM docker.io/library/ruby:2.7.5-slim-buster@sha256:53414fd155eba7e7bebe6d781fab7646b705e42dc9ed1c5be0217b85faeff77  0.0s
 => CACHED [ 2/14] RUN mkdir -p /docker-files                                                                                     0.0s
 => CACHED [ 3/14] COPY .docker-files/ /docker-files                                                                              0.0s
 => CACHED [ 4/14] RUN chmod +x /docker-files/*.sh                                                                                0.0s
 => ERROR [ 5/14] RUN /docker-files/deps.sh                                                                                     112.9s

 > [ 5/14] RUN /docker-files/deps.sh:
#0 0.541 + apt-get update -qq
#0 112.8 W: Failed to fetch http://deb.debian.org/debian/dists/buster/InRelease  Temporary failure resolving 'deb.debian.org'
#0 112.8 W: Failed to fetch http://security.debian.org/debian-security/dists/buster/updates/InRelease  Temporary failure resolving 'security.debian.org'
#0 112.8 W: Failed to fetch http://deb.debian.org/debian/dists/buster-updates/InRelease  Temporary failure resolving 'deb.debian.org'
#0 112.8 W: Some index files failed to download. They have been ignored, or old ones used instead.
#0 112.8 + DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends build-essential gnupg2 curl less git vim imagemagick zlib1g-dev
#0 112.8 Reading package lists...
#0 112.8 Building dependency tree...
#0 112.8 Reading state information...
#0 112.8 Package gnupg2 is not available, but is referred to by another package.
#0 112.8 This may mean that the package is missing, has been obsoleted, or
#0 112.8 is only available from another source
#0 112.8 However the following packages replace it:
#0 112.8   gpgv
#0 112.8
#0 112.8 Package less is not available, but is referred to by another package.
#0 112.8 This may mean that the package is missing, has been obsoleted, or
#0 112.8 is only available from another source
#0 112.8
#0 112.8 E: Unable to locate package build-essential
#0 112.8 E: Package 'gnupg2' has no installation candidate
#0 112.8 E: Unable to locate package curl
#0 112.8 E: Package 'less' has no installation candidate
#0 112.8 E: Unable to locate package git
#0 112.8 E: Unable to locate package vim
#0 112.8 E: Unable to locate package imagemagick
------
failed to solve: executor failed running [/bin/sh -c /docker-files/deps.sh]: exit code: 100
</div>

**Comment by michelson** [comment]
it seems that you got a temporary connection issue as the error trace states:

```
#0 112.8 W: Failed to fetch http://deb.debian.org/debian/dists/buster/InRelease Temporary failure resolving 'deb.debian.org'
#0 112.8 W: Failed to fetch http://security.debian.org/debian-security/dists/buster/updates/InRelease Temporary failure resolving 'security.debian.org'
#0 112.8 W: Failed to fetch http://deb.debian.org/debian/dists/buster-updates/InRelease Temporary failure resolving 'deb.debian.org'
```

**Comment by TuxADN** [comment]
Thanks for your response Miguel.

After more than 24 hours still the same, and I was able to download from http://deb.debian.org/debian/dists/buster-updates/InRelease so there it was something more.
I was able to resolve it by adding a DNS IP address in the /etc/docker/daemon.json file as follows and then rebooted the server.

```
{ 
  "dns": ["8.8.8.8"] 
}
```

Ref:
https://stackoverflow.com/questions/61567404/docker-temporary-failure-resolving-deb-debian-org
